sphinx/user-guide/installation.rst

```diff
@@ -0,0 +1,427 @@
+given repository base name NAME (e.g. ``roses_foo``), change into a
+new directory and enter the following commands::
+
+   svn co -q "svn://${HOST_AND_PATH}/${NAME}/"
+   svn ps fcm:layout -F - "${NAME}" <<'__FCM_LAYOUT__'
+   depth-project = 5
+   depth-branch = 1
+   depth-tag = 1
+   dir-trunk = trunk
+   dir-branch =
+   dir-tag =
+   level-owner-branch =
+   level-owner-tag =
+   template-branch =
+   template-tag =
+   __FCM_LAYOUT__
+   svn ci -m 'fcm:layout: defined.' "${NAME}"
+   rm -fr "${NAME}"
+
+Add the following hook scripts to the repository:
+
+* pre-commit::
+
+     #!/bin/bash
+     exec /path/to/rose/sbin/rosa svn-pre-commit "$@"
+
+* post-commit::
+
+      #!/bin/bash
+      exec /path/to/rose/sbin/rosa svn-post-commit "$@"
+
+You should replace ``/path/to/rose/`` with the location of your Rose
+installation.
+
+Make sure the hook scripts are executable.
+
+The ``rosa svn-post-commit`` command in the ``post-commit`` hook is used
+to populate a database with the suite discovery information as suites
+are committed to the repository. Edit the ``[rosie-db]`` settings
+in ``etc/rose.conf`` to point to your host machine and provide relevant
+paths such as the location for your repository and database.
+
+Once you have done that, create the Rosie database by running::
+
+   /path/to/rose/sbin/rosa db-create
+
+Make sure that the account that runs the repository hooks has read/write
+access to the database and database directory.
+
+You can test that everything is working using the built-in web server.
+Edit the ``[rosie-ws]`` settings in ``etc/rose.conf`` to configure
+the web server's log directory and port number. Start the web server
+by running::
+
+   setsid /path/to/rose/bin/rosie disco start 0</dev/null 1</dev/null 2>&1 &
+
+Check that the server is up and running using ``curl`` or a local
+web browser. E.g. If you have configured the server's port to be 1234,
+you can do::
+
+   curl -I http://localhost:1234/
+
+It should return a HTTP code 200.
+
+Alternatively you can run the Rosie web service under Apache ``mod_wsgi``.
+To do this you will need to set up an Apache module configuration file
+(typically in ``/etc/httpd/conf.d/rose-wsgi.conf``) containing the
+following (with the paths set appropriately)::
+
+   WSGIPythonPath /path/to/rose/lib/python
+   WSGIScriptAlias /rosie /path/to/rose/lib/python/rosie/ws.py
+
+Use the Apache log at e.g. ``/var/log/httpd/`` to debug problems.
+See also Configuring Rose Bush below.
+
+.. TODO ---
+
+Hopefully, you should now have a working Rosie service server. Configure
+the client settings by editing the ``[rosie-id]`` settings in
+``etc/rose.conf``. If you are using the built-in web server, you
+should ensure that you include the port number in the URL. E.g.:
+
+.. code-block:: rose
+
+   [rosie-id]
+   prefix-ws.foo=http://127.0.0.1:1234/foo
+
+You should now be able to talk to the Rosie web service server using
+the Rosie web service client. Test by doing::
```

Comment:
'executing' (or similar) instead of 'doing' (with the same applying for line 392)? 'doing' sounds a bit clumsy.